src/main/scala/beam/agentsim/agents/household/HouseholdCAVScheduling.scala

```diff
@@ -1,195 +1,398 @@
+        val dest = wayPoints(1)
+        val origin = SpaceTime(orig.activity.getCoord, math.round(orig.activity.getEndTime).toInt)
+        val routingRequest = RoutingRequest(
+          orig.activity.getCoord,
+          dest.activity.getCoord,
+          origin.time,
+          IndexedSeq(),
+          IndexedSeq(
+            StreetVehicle(Id.create(cav.id.toString, classOf[Vehicle]), cav.beamVehicleType.id, origin, CAV, true)
+          )
+        )
+        newMobilityRequests = orig.copy(routingRequestId = Some(routingRequest.requestId)) +: newMobilityRequests
+        Some(routingRequest)
+      }
+      .toList
+    (requestList, new CAVSchedule(newMobilityRequests, cav, occupancy))
+  }
+  // ***
+  override def toString = {
+    schedule
+      .foldLeft(new StringBuilder) {
+        case (output, request) => {
+          output.insert(0, "\t\t" + request.toString + "\n")
+        }
+      }
+      .insert(0, s"\tcav-id:${cav.id}\n")
+      .toString
+  }
+}
 
-    // extract potential household CAV requests from plans
-    val householdRequests = new HouseholdPlansToMSR(plans);
+object HouseholdCAVScheduling {
 
-    // set up the household fleet
-    for (i <- 0 until fleetSize) {
-      fleet = fleet :+ HouseholdCAV(i, 4) //TODO probably to be extracted from Attr Map
-    }
+  def computeSkim(
+    plans: List[BeamPlan],
+    avgSpeed: Map[BeamMode, Double]
+  ): Map[BeamMode, Map[Coord, Map[Coord, Double]]] = {
+    val activitySet: Set[Coord] = (for {
+      plan <- plans
+      act  <- plan.activities
+    } yield act.getCoord).toSet
 
-    // deploy the fleet or set up the initial household schedule
-    var emptyFleetSchedule = List[CAVSchedule]()
-    fleet.foreach(
-      x =>
-        emptyFleetSchedule = emptyFleetSchedule :+ new CAVSchedule(
-          List[MobilityServiceRequest](
-            new MobilityServiceRequest(
-              None,
-              householdRequests.requests.head.activity,
-              householdRequests.requests.head.time - 1,
-              1,
-              MSRInit
-            )
-          ),
-          x,
-          0,
-          0
-      ) // initial Cost and Occupancy
+    var skim: Map[BeamMode, Map[Coord, Map[Coord, Double]]] = Map(
+      BeamMode.CAR     -> Map[Coord, Map[Coord, Double]](),
+      BeamMode.TRANSIT -> Map[Coord, Map[Coord, Double]]()
     )
-    feasibleSchedules = feasibleSchedules :+ new HouseholdSchedule(emptyFleetSchedule)
-
-    // extract all possible schedule combinations
-    for (request <- householdRequests()) {
-      var householdSchedulesToAdd = List[HouseholdSchedule]()
-      var householdSchedulesToDelete = List[HouseholdSchedule]()
-      for (schedule <- feasibleSchedules) {
-        householdSchedulesToAdd ++= schedule.check(request)
-        if (!schedule.feasible) {
-          householdSchedulesToDelete = householdSchedulesToDelete :+ schedule
-        }
+    for (src <- activitySet;
+         dst <- activitySet) yield {
```

Comment:
Same as above - if you are going for mutation, then you don't want a yield (map/flatMap), but instead a foreach (no yield)